.github/workflows/ci-pipeline.yml

```diff
@@ -46,3 +46,52 @@ jobs:
 
       - name: Deploy to Azure Web App
         run: curl -dH -X POST ${{ secrets.AZURE_WEB_HOOK }}
+
+  azure-deploy:
+    name: deploy to Azure Web App
+    runs-on: ubuntu-latest
+    needs: docker-publish
+    steps:
+      - name: Deploy to Azure Web App
+        run: curl -dH -X POST ${{ secrets.AZURE_WEB_HOOK }}
+
+
+
+  terraform-deploy:
+    name: deploy using terraform
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged
+
+    defaults:
+      run:
+        working-directory: ./terraform
+
+    env:
+      ARM_ACCESS_KEY: ${{ secrets.ARM_ACCESS_KEY }}
+      ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+      ARM_CLIENT_SECRET: ${{ secrets.ARM_CLIENT_SECRET }}
+      ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+      ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+      TF_VAR_OAUTH_CLIENT_ID: ${{ secrets.OAUTH_CLIENT_ID }}
+      TF_VAR_OAUTH_CLIENT_SECRET: ${{ secrets.OAUTH_CLIENT_SECRET }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_wrapper: false
+
+      - run: terraform fmt
```

Comment:
Running this here doesn't achieve a lot; it formats some files that will never be read by a human! Changing this to add the `-check` flag might be what you're looking for - that way it will fail the pipeline if the files aren't correctly formatted (but personally I'd suggest that having that as an extra step in the `build-test` job would make a lot more sense - we want to know about this _before_ we actually try to deploy!)